src/main/java/uk/ac/ic/wlgitbridge/snapshot/base/SnapshotAPIRequest.java

```diff
@@ -33,27 +30,11 @@ protected void onBeforeRequest(
     ) throws IOException {
         if (oauth2 != null) {
             request.setInterceptor(request1 -> {
-                new BasicAuthentication(
-                        USERNAME,
-                        PASSWORD
-                ).intercept(request1);
                 oauth2.intercept(request1);
             });
-        } else {
```

Comment:
Nah, I think what this code used to do was:

- if oauh2
  - intercept with basic auth
  - intercept with oauth
- else
  - intercept with basic auth


Which boils down to:

- intercept with basic auth
- if oauh2
  - intercept with oauth


And in this PR we're trimming it down to:

- if oauh2
  - intercept with oauth

Reply:
I've pushed a commit to remove `IOException` from the signature :+1:

Reply:
Another way of looking at this might be... will `oauth2` ever be null? I don't think there are any endpoints in the API that wouldn't take oauth2 credentials, so we could remove the null check here --- it would be a bug to not specify a credential.

Reply:
Gotcha, I'll put in a `RuntimeException` when this is null :+1:

Reply:
Sigh, this is unfortunately not the case.

The OAuth2Filter passes `oauth2` as null, when checking if authentication is required? https://github.com/overleaf/writelatex-git-bridge/blob/master/src/main/java/uk/ac/ic/wlgitbridge/server/Oauth2Filter.java#L104-L105

So it seems there are parts of the code that _do_ rely on this being null at certain points.

Reply:
Right, good find. I think that might also be a throwback to when we supported anonymous access. The endpoint it's trying to hit always requires OAuth, so I think it is always getting a 401 and then trying again with the credentials. That's what it looks like in the logs (searched for `"api/v0/docs"`). Probably we should fix that... but it would be expanding the scope a bit here. WDYT?

Reply:
Anonymous access strikes again! :D

I'll take a swing at it, and see if it can be stripped out easily, as it would be good to be rid of this code :+1:

Reply:
Ugh, I think the problem now is that the integration tests all run in this anonymous mode, and so with the change they all just pause and prompt for username/password input :(

Reply:
I think we may have to punt this to a later ticket

Reply:
Here's an new ticket to rip out the last traces of anonymous mode: https://github.com/overleaf/issues/issues/4377